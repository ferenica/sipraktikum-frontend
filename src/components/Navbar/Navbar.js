import React from 'react';
import PropTypes from 'prop-types';
import makaraui from '../../assets/INOS/Makara_UI.png';
import ProfileBtn from './ProfileBtn/ProfileBtn';

import {makeStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import MuiButton from '@material-ui/core/Button';


const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 0,
  },
  viewButtonLabel: {
    textTransform: 'none',
  },
  appBar: {
    position: 'static',
    color: '#404852',
    backgroundColor: '#ffffff',
  },
  avatar: {
    marginRight: 15,
  },
  typography: {
    variant: 'body1',
    flexGrow: 1,
  },
  buttomNavigation: {
    fontSize: 20,
    flexGrow: 1,
  },
  button: {
    'borderRadius': '20px',
    'textTransform': 'none',
    'letterSpacing': '0',
    'fontFamily': 'Nunito Sans',
    'fontWeight': 'bold',

    'backgroundColor': '#FF8326',
    'color': 'white',
    '&:hover': {
      backgroundColor: '#FFFFFF',
      color: '#FF8326',

    },
  },
  icon: {
    width: '1.5em',
  },
}));

/**
 * Navbar
 * @param {Boolean} props - For login status
 * @return {Component} Navbar
 */
export default function Navbar(props) {
  const classes = useStyles();

  const handleLogin = () => {
    window.location.replace('/login');
  };

  let navigation;
  let button;

  if (props.isAdmin) {
    navigation = <BottomNavigation showLabels>
      <BottomNavigationAction
        className={classes.buttomNavigation}
        label="Home"
        href="/"/>
      <BottomNavigationAction
        className={classes.buttomNavigation}
        label="Kelola Penugasan"
        href="/admin-penugasan-dosen"/>
      <BottomNavigationAction
        className={classes.buttomNavigation}
        label="Statistik"
        href="/statistik" />
      <BottomNavigationAction
        className={classes.buttomNavigation}
        label="Laporan Akhir"
        href="/laporan-akhir" />
    </BottomNavigation>;
  } else if (props.isDosen) {
    navigation = <BottomNavigation showLabels>
      <BottomNavigationAction
        className={classes.buttomNavigation}
        label="Home"
        href="/"/>
      <BottomNavigationAction
        className={classes.buttomNavigation}
        label="Arsip Laporan"
        href="/arsip-laporan"/>
    </BottomNavigation>;  
  } else {
    navigation = <BottomNavigation showLabels>

      <BottomNavigationAction
        className={classes.buttomNavigation}
        label="Home"
        href="/"/>
    </BottomNavigation>;
  }

  if (props.isAuthenticated) {
    button = <ProfileBtn/>;
  } else {
    button =
    <div>
      <MuiButton
        data-testid="profbtn"
        className={classes.button}
        disableElevation
        variant="contained"
        aria-haspopup="true"
        onClick={handleLogin}>
        Login
      </MuiButton>
    </div>;
  }

  return (
    <div>
      <AppBar className={classes.appBar}>
        <Toolbar>
          <Avatar className={classes.avatar} src={makaraui}></Avatar>
          <Typography className={classes.typography}>
            <strong>Sistem Informasi Penilaian & Database Praktikum</strong>
            <br></br>
            Fakultas Ilmu Sosial dan Ilmu Politik Universitas Indonesia
          </Typography>
          {navigation}
          {button}
        </Toolbar>
      </AppBar>
    </div>
  );
}

Navbar.propTypes = {
  isAdmin: PropTypes.bool,
  isDosen: PropTypes.bool,
  isAuthenticated: PropTypes.bool,
};
