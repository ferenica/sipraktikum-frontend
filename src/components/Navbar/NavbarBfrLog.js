import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import makaraui from '../../assets/INOS/Makara_UI.png';
import MuiButton from '@material-ui/core/Button';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 0,
  },
  viewButtonLabel: {
    textTransform: 'none',
  },
  appBar: {
    position: 'static',
    color: '#404852',
    backgroundColor: '#ffffff',
  },
  avatar: {
    marginRight: 15,
  },
  typography: {
    variant: 'body1',
    flexGrow: 1,
  },
  buttomNavigation: {
    fontSize: 20,
    flexGrow: 1,
  },
  button: {
    'borderRadius': '20px',
    'textTransform': 'none',
    'letterSpacing': '0',
    'fontFamily': 'Nunito Sans',
    'fontWeight': 'bold',

    'backgroundColor': '#FF8326',
    'color': 'white',
    '&:hover': {
      backgroundColor: '#FFFFFF',
      color: '#FF8326',

    },
  },
  icon: {
    width: '1.5em',
  },
}));
/**
 *
 */
export default function NavbarBfrLog() {
  const classes = useStyles();
  const [setAnchorEl] = React.useState(null);

  return (
    <div>
      <AppBar className={classes.appBar}>
        <Toolbar>
          <Avatar className={classes.avatar} src={makaraui}></Avatar>
          <Typography className={classes.typography}>
            <strong>Sistem Database Praktikum</strong><br></br>
            Fakultas Ilmu Sosial dan Ilmu Politik Universitas Indonesia
          </Typography>

          <MuiButton
            data-testid = "profbtn"
            className = {classes.button}
            disableElevation
            variant="contained"
            aria-haspopup="true"
            href = "http://ppl-berkah-backend.herokuapp.com/auth/login/sivitas/">
            <AccountCircleIcon className = {classes.icon} fontSize="small"/>
            Login

          </MuiButton>
        </Toolbar>
      </AppBar>
    </div>
  );
}
