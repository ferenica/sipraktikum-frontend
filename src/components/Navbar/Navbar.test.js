import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import Navbar from './Navbar';

Enzyme.configure({adapter: new EnzymeAdapter()});

describe('Navbar', () => {
  const component = shallow(<Navbar />);
  it('should renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
