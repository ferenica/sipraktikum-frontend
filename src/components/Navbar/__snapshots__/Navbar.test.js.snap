// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should renders correctly 1`] = `ShallowWrapper {}`;
